python/semantic_kernel/planning/basic_planner.py

```diff
@@ -0,0 +1,208 @@
+
+        # Create a mapping between all function names and their descriptions
+        # and also a mapping between function names and their parameters
+        all_functions = native_functions
+        skill_names = list(all_functions.keys())
+        all_functions_descriptions_dict = {}
+        all_functions_params_dict = {}
+
+        for skill_name in skill_names:
+            for func in all_functions[skill_name]:
+                key = skill_name + "." + func.name
+                all_functions_descriptions_dict[key] = func.description
+                all_functions_params_dict[key] = func.parameters
+
+        # Create the [AVAILABLE FUNCTIONS] section of the prompt
+        available_functions_string = ""
+        for name in list(all_functions_descriptions_dict.keys()):
+            available_functions_string += name + "\n"
+            description = all_functions_descriptions_dict[name]
+            available_functions_string += "description: " + description + "\n"
+            available_functions_string += "args:\n"
+
+            # Add the parameters for each function
+            parameters = all_functions_params_dict[name]
+            for param in parameters:
+                if not param.description:
+                    param_description = ""
+                else:
+                    param_description = param.description
+                available_functions_string += (
+                    "- " + param.name + ": " + param_description + "\n"
+                )
+            available_functions_string += "\n"
+
+        return available_functions_string
+
+    async def create_plan_async(
+        self,
+        goal: str,
+        kernel: Kernel,
+        prompt: str = PROMPT,
+    ) -> Plan:
+        """
+        Creates a plan for the given goal based off the functions that
+        are available in the kernel.
+        """
+
+        # Create the semantic function for the planner with the given prompt
+        planner = kernel.create_semantic_function(
+            prompt, max_tokens=1000, temperature=0.8
+        )
+
+        available_functions_string = self._create_available_functions_string(kernel)
+
+        # Create the context for the planner
+        context = ContextVariables()
+        # Add the goal to the context
+        context["goal"] = goal
+        context["available_functions"] = available_functions_string
+        generated_plan = await planner.invoke_async(variables=context)
+        return Plan(prompt=prompt, goal=goal, plan=generated_plan)
+
+    async def execute_plan_async(self, plan: Plan, kernel: Kernel) -> str:
+        """
+        Given a plan, execute each of the functions within the plan
+        from start to finish and output the result.
+        """
+        generated_plan = json.loads(plan.generated_plan.result)
+
+        context = ContextVariables()
+        context["input"] = generated_plan["input"]
+        subtasks = generated_plan["subtasks"]
```

Comment:
again, could be part of the Plan() initialization, but doesn't have to be addressed in this PR

Reply:
Let's do this in a future PR!